components/youtube_script_injector/browser/core/youtube_component_installer.cc

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/youtube_script_injector/browser/core/youtube_component_installer.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/functional/callback_forward.h"
+#include "base/no_destructor.h"
+#include "brave/components/brave_component_updater/browser/brave_on_demand_updater.h"
+#include "brave/components/youtube_script_injector/browser/core/youtube_registry.h"
+#include "brave/components/youtube_script_injector/common/features.h"
+#include "components/component_updater/component_installer.h"
+#include "components/component_updater/component_updater_service.h"
+#include "crypto/sha2.h"
+
+namespace youtube_script_injector {
+
+namespace {
+
+// Directory structure of YouTube Script Injector component:
+// lhhcaamjbmbijmjbnnodjaknblkiagon/<component version>/
+//  |_ manifest.json
+//  |_ youtube.json
+//  |_ scripts/
+//    |_ keep-playing-audio.js
+//    |_ fullscreen.js
+// See youtube_json.cc for the format of youtube.json.
```

Comment:
I was trying to locate where the component would be. I saw some branches in
- https://github.com/brave/brave-core-crx-packager/compare/master...youtube_fullscreen_landscape
- https://github.com/brave/brave-core-crx-packager/compare/master...youtube_quality_android

But they didn't seem to be packaging a component. Looking at this extension ID (`lhhcaamjbmbijmjbnnodjaknblkiagon`), it seems we would be piggybacking off the [`Brave Privacy Settings Selection for Sites Tool (PSST) Files` component](https://github.com/brave/brave-core/blob/e26734efeb0a24b767e32741517b1f45ca4a38a9/components/psst/browser/core/psst_component_installer.cc#L42-L53). That seems to be guarded by the brave://flags/#brave-psst flag (default OFF).

@ShivanKaul does it make sense to club these YouTube specific scripts into there?

I also couldn't find any packaging for PSST. Usually, you'd see something in https://github.com/brave/brave-core-crx-packager which would include an `npm run` command. I checked the job list in ci.brave.com and didn't see anything either.

Reply:
I'll include a better and more elaborate answer tomorrow: I've just picked a temporary extension ID for testing. Next step will be creating a new one specifically for the youtube injector component before marking this PR as ready to review.